Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -716,7 +717,10 @@ public async Task Save()
 
         var rowSavingHandler = editState == DataGridEditState.New ? RowInserting : RowUpdating;
 
-        if ( await IsSafeToProceed( rowSavingHandler, editItem, editedCellValues ) )
+        var editItemClone = editItem.DeepClone();
+        SetItemEditedValues( editItemClone );
+
+        if ( await IsSafeToProceed( rowSavingHandler, editItem, editItemClone, editedCellValues ) )

```

Comment:
Should we also close the `OldItem` to prevent users from accidentally changing the original object values?

Reply:
Can you elaborate?

Reply:
When user receives an eventArgument through the callback they could override one of the values

eg.

```cs
Task OnUpdating(eventArgs)
{
   eventArgs.Item.Name = "new name";
}
```

Reply:
Well, that always has been like that. What do you propose? Even the `OldItem` to be cloned?

Reply:
It was. We only added `.Item` because users kept requesting it. And I was always reluctant to add it because it could interfere with a dictionary of editing Values. If we clone that object then there would be no danger of overriding anything in the original Data.

Reply:
I see, well we can add the clone there too I guess. protecting users from themselves. 🗡️

Reply:
Another note, is that cloning the .Item will be a breaking change for whover who was using it as a reference. Are we okay with this? 
**Note to self:** Update docs with this.

Reply:
> Another note, is that cloning the .Item will be a breaking change for whover who was using it as a reference. Are we okay with this?

Damn, I need to think about this.

Reply:
Honestly, I would prefer if we didn't clone it. Having the reference still holds some utility.

We'll be removing `.Item` in the future, so users will be less likely to try and mutate something called `OldItem`. And anyway, if they do that, they kinda do it at their own risk, can't protect every possible miss-usage...

Reply:
🤔 OK, then leave the Item as before. Better not to break anything in 1.x.

Reply:
By utility, I mean, something like this:
![image](https://github.com/Megabit/Blazorise/assets/22283161/53a0f5fc-33e8-4a96-b452-ae3a658ff2fe)
And maybe other use cases which can require the reference.